app/controllers/organizations_controller.rb

```diff
@@ -2,7 +2,11 @@ class OrganizationsController < ApplicationController
   before_action :load_resource, only: [:show, :edit, :update, :set_current]
 
   def index
-    @organizations = Organization.all.page(params[:page]).per(25)
+    @organizations = if params[:q].present?
+                       Organization.search_by_query(params[:q]).page(params[:page]).per(25)
+                     else
+                       Organization.all.page(params[:page]).per(25)
+                     end
```

Comment:
what about something like?

```
organizations  = Organization.all
organizations  = organizations.search_by_query(params[:q]) if params[:q].present?
@organizations = organizations.page(params[:page]).per(25)
```

Reply:
I think the efficiency of the algorithm you propose is worse, you always make a call to fetch all organisations even if the "important" query only fetches two, right?

Reply:
Not really, in ActiveRecord those queries are "lazy" evaluated, the first time a `to_a` or similar (`each` in the view) is invoked, so meanwhile you can keep chaining new scopes and/or filters.

Reply:
oh I didn't know, thanks for the explanation :)

Reply:
You can check the log to see where and how each query is exactly triggered.